docs/DeveloperGuide.adoc

```diff
@@ -297,86 +299,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 [discrete]
-=== Use case: Delete person
+=== Use Case 1: Add Problem
+
+*Guarantees*
+
+Problem will be added to the application if and only if the required fields are specified and the details entered are all valid.
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User launches AlgoBase.
+2.  User adds a new problem by entering the name of the problem, optionally specifying the description, author, weblink, source as well as any remarks or tags.
+3.  AlgoBase indicates successful addition of new problem.
+4.  AlgoBase displays details of problem added.
 +
 Use case ends.
 
 *Extensions*
+[none]
+* 2a. AlgoBase detects that an existing name already exists.
++
+[none]
+** 2a1. AlgoBase informs user that problem was not successfully added because the name already exists.
++
+Use case ends.
+
+* 2b. AlgoBase detects a missing name or invalid format for some field(s).
++
+[none]
+** 2b1. AlgoBase informs user that problem was not successfully added because the format is invalid.
++
+Use case ends.
+
+[discrete]
+=== Use Case 2: Edit Problem
+
+*Preconditions*
+
+1.  There are existing problems in the storage of AlgoBase.
+2.  The command is formatted correctly and the provided arguments meet the requirement for corresponding fields.
+
+*Guarantees*
+
+The matched problem in the application will be edited with the provided fields.
+
+*MSS*
 
+1.  User launches AlgoBase.
+2.  User edits an existing problem by entering the index, followed by fields that the user intends to edit (including name, description, weblink, author, source, remark, tag, etc.).
+3.  AlgoBase indicates successful edition of the existing problem.
+4.  AlgoBase updates the UI with the updated problem.
++
+Use case ends.
+
+*Extensions*
 [none]
-* 2a. The list is empty.
+* 2a. AlgoBase detects the index is out of bounds.
++
+[none]
+** 2a1. AlgoBase informs user that the edition is unsuccessful because the index is out of bounds.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+* 2b. AlgoBase detects no provided fields.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 2b1. AlgoBase informs user that nothing is updated.
 +
-Use case resumes at step 2.
+Use case ends.
 
-_{More to be added}_
+[discrete]
+=== Use Case 3: Delete Problem
 
-[appendix]
-== Non Functional Requirements
+*Guarantees*
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+Problem will be deleted and removed from local storage of AlgoBase and cannot find from relevant tags.
```

Comment:
If you are removing from local storage here, should you be mentioning the adding and editing of problems into local storage above as well?